cli/pkg/local/app.go

```diff
@@ -475,3 +489,12 @@ func ParseLogFormat(format string) (LogFormat, bool) {
 		return "", false
 	}
 }
+
+func scanPort(port int) error {
+	conn, err := net.Listen("tcp", ":"+strconv.Itoa(port))
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	return nil
+}
```

Comment:
1. Rename to `checkPort`
2. Do you have any resources on whether it's good practice to check ports using `Listen`? Sometimes I feel it takes the system a tiny bit of time to acquire/close a port. Might be better to try a dial?